crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -131,35 +121,64 @@ impl Cursor {
 
     /// Returns the currently pointed to [`Node`].
     pub fn node(&self) -> Node {
-        self.current.node.clone()
+        self.node.clone()
     }
 
     pub fn node_name(&self) -> Option<FieldName> {
-        self.path.last().and_then(|parent| {
-            let this = &parent.rule_node.children[self.current.child_number];
+        self.parent.as_ref().and_then(|parent| {
+            let this = &parent.rule_node.children[self.child_number];
 
             this.name
         })
     }
 
     /// Returns the text offset that corresponds to the beginning of the currently pointed to node.
     pub fn text_offset(&self) -> TextIndex {
-        self.current.text_offset
+        self.text_offset
     }
 
     /// Returns the text range that corresponds to the currently pointed to node.
     pub fn text_range(&self) -> TextRange {
-        self.current.text_range()
+        let start = self.text_offset;
+        let end = start + self.node.text_len();
+        start..end
     }
 
     /// Returns the depth of the current node in the CST, i.e. the number of ancestors.
     pub fn depth(&self) -> usize {
-        self.path.len()
+        let mut depth = 0;
+        if let Some(parent) = &self.parent {
+            let mut parent = parent;
+            depth += 1;
+            while let Some(grandparent) = &parent.parent {
+                depth += 1;
+                parent = grandparent;
+            }
+        }
+
+        depth
     }
 
-    /// Returns an iterator over the current node's ancestors, starting from the cursor root node.
-    pub fn ancestors(&self) -> impl Iterator<Item = &Rc<RuleNode>> {
-        self.path.iter().map(|elem| &elem.rule_node)
+    /// Returns an iterator over the current node's ancestors, starting from the parent of the current node.
+    pub fn ancestors(&self) -> impl Iterator<Item = Rc<RuleNode>> {
```

Comment:
It's worth noting that this change reverted the order of the ancestors, which is what I hit when upgrading hardhat-vscode to Slang 0.14 in https://github.com/NomicFoundation/hardhat-vscode/pull/554.

@OmarTawfik @AntonyBlakey do we plan to keep a "migration guide" between select versions or should we only care starting from v1?

Reply:
I think the `changeset` markdown is intended to serve as a light-weight migration guide.
I wonder if we can add multiple lines/links/images/etc... of markdown to each changeset, and how it will be rendered in the final release changelog/announcement:

- https://github.com/NomicFoundation/slang/releases/tag/v0.14.0

Do you think that wouldn't be enough? IMO, a separate migration guide might be more beneficial once the API is stabilized and we have an official v1.

Reply:
I think the `changeset` markdown can double as a light-weight migration guide:

- https://github.com/NomicFoundation/slang/releases/tag/v0.14.0

Do you think that wouldn't be enough? IMO, a separate migration guide might be more beneficial once the API is stabilized and we have an official v1.